README.md

```diff
@@ -1 +1,412 @@
+
+We can compare these two roles to the phases of constructing a building: the `alloc` method is like selecting the ground that will be the building's lot, designating and preparing the place for it to go. The `init` method, however, is the phase of actually building the structure that's detailed in the architect's blueprint.
+
+The `init` method is actually just the basic initializer, called the "default initializer", but an initializer is any method that returns an `instancetype` and the compiler will require that the method name begins with `init...`. Methods from the `init` family can be set up to offer varying levels of pre-fabricated setup. 
+
+These "convenience initializers" and "designated initializers" can be seen like asking the construction firm to paint the house something other than white, or even to find an interior designer to provide furniture and to hang art on the walls. In such a case, the house is further along to becoming an actual home instead of just an empty free-standing structure that serves as the most basic form of a house.
+
+#### Initializing Read-Only Values
+
+Initializers are necessary for setting up read-only properties that cannot be written to from outside the class, such as in the case of creating an instance of `NSSortDescriptor`:
+
+```objc
+// incorrect: assigning to read-only properties
+
+NSSortDescriptor *sortByNameAsc = [[NSSortDescriptor alloc] init];
+sortByNameAsc.key = @"name";
+sortByNameAsc.ascending = YES;
+```
+
+![](https://curriculum-content.s3.amazonaws.com/ios-intro-to-objects-unit/readonly_error.png)
+
+The initializer allows a user of that class to provide an initial value for those read-only properties upon creating an instance:
+
+```objc
+// correct: using an initializer to give the properties initial values
+
+NSSortDescriptor *sortByNameAsc = [[NSSortDescriptor alloc] initWithKey:@"name"
+                                                              ascending:YES];
+```
+
+![](https://curriculum-content.s3.amazonaws.com/ios-intro-to-objects-unit/readonly_init.png)
+
+The usefulness of providing an initializer allows the user of the class to be assured that the instance will be set up holistically enough to perform its methods without unexpected behavior. It also allows us, the developer, to document the set up for the class.
+
+### The Designated Initializer
+
+A "designated initializer" is the initializer method which provides the *most thorough coverage of that class's data.* It doesn't necessarily provide *complete* coverage, just the *most* coverage of any of the `init...` methods. Take for example, the `initWithObjects:` method available to `NSMutableArray`.
```

Comment:
It looks like the discussion of that method doesn't match the table of contents, which you're right. This is what I was making that claim based on:

https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Classes/NSArray_Class/#//apple_ref/occ/instm/NSArray/initWithObjects:

Discussion
After an immutable array has been initialized in this way, it can’t be modified.

This method is a designated initializer.